peps/pep-0734.rst

```diff
@@ -0,0 +1,400 @@
+  finalized.  Thus no interpreters created through
+  ``interpreters.create()`` will leak.
+
+|
+
+* ``Interpreter.is_running()`` refers only to if there is a thread
+  running a script (code) in the interpreter's ``__main__`` module.
+  That basically means whether or not ``Interpreter.exec()`` is running
+  in some thread.  Code running in sub-threads is ignored.
+
+|
+
+* ``Interpreter.set_main_attrs()`` will only allow (for now) objects
+  that are specifically supported for passing between interpreters.
+  See `Shareable Objects`_.
+
+|
+
+* ``Interpreter.set_main_attrs()`` is helpful for initializing the
+  globals for an interpreter before running code in it.
+
+|
+
+* ``Interpreter.exec()`` does not reset the interpreter's state nor
+  the ``__main__`` module, neither before nor after, so each
+  successive call picks up where the last one left off.  This can
+  be useful for running some code to initialize an interpreter
+  (e.g. with imports) before later performing some repeated task.
+
+Comparison with builtins.exec()
+-------------------------------
+
+``Interpreter.exec()`` is essentially the same as the builtin
+``exec()``, except it targets a different interpreter, using that
+interpreter's isolated state.
+
+The builtin ``exec()`` runs in the current OS thread and pauses
+whatever was running there, which resumes when ``exec()`` finishes.
+No other threads are affected.  (To avoid pausing the current thread,
+run ``exec()`` in a ``threading.Thread``.)
+
+``Interpreter.exec()`` works the same way.
+
+The builtin ``exec()`` take a namespace against which it executes.
+It uses that namespace as-is and does not clear it before or after.
+
+``Interpreter.exec()`` works the same way.
+
+...with one slight difference: the namespace is implicit
+(the ``__main__`` module's ``__dict__``).  This is the same as how
+scripts run from the Python commandline or REPL work.
+
+The builtin ``exec()`` discards any object returned from the
+executed code.
+
+``Interpreter.exec()`` works the same way.
+
+The builtin ``exec()`` propagates any uncaught exception from the code
+it ran.  The exception is raised from the ``exec()`` call in the
+thread that originally called ``exec()``.
+
+``Interpreter.exec()`` works the same way.
+
+...with one slight difference.  Rather than propagate the uncaught
+exception directly, we raise an ``interpreters.RunFailedError``
+with a snapshot of the uncaught exception (including its traceback)
+as the ``__cause__``.  Directly raising (a proxy of) the exception
+is problematic since it's harder to distinguish between an error
+in the ``Interpreter.exec()`` call and an uncaught exception
+from the subinterpreter.
+
+API: Communicating Between Interpreters
+---------------------------------------
+
+The module introduces a basic communication mechanism called "channels".
+They are based on `CSP`_, as is ``Go``'s concurrency model (loosely).
+Channels are like pipes: FIFO queues with distinct send/recv ends.
```

Comment:
?
```suggestion
Channels are like pipes: FIFO queues with distinct send/receive ends.
```

Reply:
I was mirroring the type and method names.  However, that doesn't add much here.  I'll spell it out.